hudi-client/hudi-java-client/src/main/java/org/apache/hudi/execution/bulkinsert/JavaCustomColumnsSortPartitioner.java

```diff
@@ -40,11 +42,14 @@
   private final String[] sortColumnNames;
   private final Schema schema;
   private final boolean consistentLogicalTimestampEnabled;
+  private final boolean isSortedByPartitionPath;
 
-  public JavaCustomColumnsSortPartitioner(String[] columnNames, Schema schema, boolean consistentLogicalTimestampEnabled) {
+  public JavaCustomColumnsSortPartitioner(String[] columnNames, Schema schema, HoodieWriteConfig config) {
+    String partitionPath = config.getString(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME);
+    this.isSortedByPartitionPath = columnNames.length > 0 && partitionPath.equals(columnNames[0]);
```

Comment:
All partition columns should match not just the first one (partition path might be composed from multiple columns)